text/2005-match-ergonomics.md

```diff
@@ -393,6 +414,26 @@ The major downside of this proposal is that it complicates the pattern-matching
 logic. However, doing so allows common cases to "just work", making the beginner
 experience more straightforward and requiring fewer manual reference gymnastics.
 
+# Future Extensions
+[future extensions]: #future_extensions
+In the future, this RFC could be extended to add support for autodereferencing
+custom smart-pointer types using the `Deref` and `DerefMut` traits.
+
+```rust
+let x: Box<i32> = Box::new(0);
```

Comment:
Something is wrong with types here. Perhaps you wanted to write the following line?
```rust
let x: Box<Option<i32>> = Box::new(Some(0));
```